.github/actions/iam-drift-detection/action.yml

```diff
@@ -47,34 +51,97 @@ runs:
+        ISSUE_LABEL: '${{ inputs.issue_label }}'
+      with:
+        result-encoding: 'string'
+        retries: '3'
+        script: |
+          const { ISSUE_LABEL } = process.env;
+          const data = (await github.rest.issues.listForRepo({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            labels: [ISSUE_LABEL],
+            state: 'all',
+          })).data[0];
+          if (data) {
+            return data.number;
+          }
+          return "";
 
-    - name: 'Create bug'
+    - uses: 'actions/github-script@d7906e4ad0b1822421a7e6a35d5ca353c962f410' # ratchet:actions/github-script@v6
+      id: 'create_issue'
       if: ${{steps.run_diff.outputs.diff_detected == 'TRUE'}}
-      # TODO: Use actions/github-script to create this issue instead of a third-party library.
-      uses: 'imjohnbo/issue-bot@3d96848fb5e9a4a473bb81ae62b4b4866a56e93a'
+      env:
+        ISSUE_LABEL: '${{ inputs.issue_label }}'
+        DIFF_FILE: '${{ steps.run_diff.outputs.diff_file }}'
+        WORKFLOW_RUN_URL: '${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}/attempts/${{ github.run_attempt }}'
       with:
-        assignees: ''
-        labels: 'IAM drift'
-        title: 'IAM drift detected'
-        body: |
-          We've detected a drift between your submitted IAM policies and actual
+        retries: '3'
+        script: |
+          const { ISSUE_LABEL, DIFF_FILE, WORKFLOW_RUN_URL } = process.env;
+          const fs = require('fs');
+          const diff_content = fs.readFileSync(DIFF_FILE, 'utf8');
+          const diff_100_lines = diff_content.split('\n').slice(0, 100).join("\n");
+          const was_shortened = diff_content !== diff_100_lines;
+          const shortened_msg = was_shortened ? '...' : '';
+          const issue_body = `We've detected a drift between your submitted IAM policies and actual
           IAM policies.
+
+          See the logs for the full details: ${WORKFLOW_RUN_URL}
           
-          ${{env.diff_snippet}}
+          ${diff_100_lines}
+          ${shortened_msg}
           
           Please determine which parts are correct, and submit updated
           terraform config and/or remove the extra policies.
           
-          Re-run this manually once complete to verify all diffs are properly resolved.
-          
-        pinned: false
-        # Given this is based on head, prior issues are obsolete or duplicates.
-        close-previous: true
+          Re-run this manually once complete to verify all diffs are properly resolved.`;
 
+          await github.rest.issues.create({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            labels: [ISSUE_LABEL],
+            title: 'IAM drift detected',
+            body: issue_body,
+          });
 
+    - uses: 'actions/github-script@d7906e4ad0b1822421a7e6a35d5ca353c962f410' # ratchet:actions/github-script@v6
+      id: 'maybe-close-previous-issue'
```

Comment:
Why this and not updating the existing one?

Reply:
Arbitrarily picked based on what it was doing before. It does seem slightly easier to close and create a new one rather than having two create paths. But I don't feel strongly either way. If you want update lets do that in a follow up?

Reply:
ACK, fine with this way.